services/search/src/search/routes/filter.py

```diff
@@ -153,6 +156,10 @@ async def filter_endpoint(request: Request) -> Response:
                         offset,
                         extensions_directory,
                     )
+                with StepProfiler(method="search_endpoint", step="clean old indexes"):
+                    # no need to do it every time
+                    if random.random() < clean_cache_proba:  # nosec
+                        clean_dir(str(duckdb_index_file_directory), expiredTimeIntervalSeconds)
```

Comment:
OK. It was the logic in /rows if I remember well.
We're profiling, so, we can assess if it impacts a lot the response time or not. But, if it does, maybe we should change the logic a bit, and run a regular job in parallel on the search pod (as is done in services/workers: the zombie killers, the long jobs killer, etc)

Reply:
Yes, it's a simple logic initially. I am estimating to retain only 12 hours of data initially, which should take at most 1 minute. Agree, in case of taking too much time, to change to a parallel process.

Reply:
> which should take at most 1 minute

it means that a user who is launching a search will have to wait 1 minute for the response, because of our maintenance operation.

If the duration were like 2 seconds, then the strategy would be OK. But if we expect regularly 1-minute durations, I think we should change the implementation, and run the maintenance operations in parallel.

Reply:
Yeah, even currently, we have requests that only take 2 minutes to answer in the index processing. Considering the slow demand for the requests, initially, it will not have a high impact on the first version of local storage. Could we handle it in another PR?

Reply:
yes, sure

Reply:
(maybe let's open an issue to keep track of it)

Reply:
Done https://github.com/huggingface/dataset-viewer/issues/2718